testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java

```diff
@@ -0,0 +1,316 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hudi;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.HUDI;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onHudi;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestHudiSparkCompatibility
+        extends ProductTest
+{
+    private static final String COW_TABLE_TYPE = "cow";
+    private static final String MOR_TABLE_TYPE = "mor";
+
+    private String bucketName;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        bucketName = requireNonNull(System.getenv("S3_BUCKET"), "Environment variable not set: S3_BUCKET");
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteShowCreateTable()
+    {
+        String tableName = "test_hudi_cow_show_create_" + randomTableSuffix();
+
+        createNonPartitionedTable(tableName, COW_TABLE_TYPE);
+
+        String showCreateTable = format(
+                "CREATE TABLE hudi.default.%s (\n" +
+                        "   _hoodie_commit_time varchar,\n" +
+                        "   _hoodie_commit_seqno varchar,\n" +
+                        "   _hoodie_record_key varchar,\n" +
+                        "   _hoodie_partition_path varchar,\n" +
+                        "   _hoodie_file_name varchar,\n" +
+                        "   id bigint,\n" +
+                        "   name varchar,\n" +
+                        "   price integer,\n" +
+                        "   ts bigint\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   location = 's3://%s/%s'\n" +
+                        ")",
+                tableName,
+                bucketName,
+                tableName);
+
+        try {
+            assertThat(onTrino().executeQuery("SHOW CREATE TABLE hudi.default." + tableName)).containsExactlyInOrder(row(showCreateTable));
+        }
```

Comment:
For Spark, the create statement will be similar to what's written in `createNonPartitionedTable` with an additional table property that shows the hudi commit time. I don't see much value in adding another test/assertion here.

Reply:
I know it will be similar to the query in `createNonPartitionedTable`. Then, how can we detect the Spark change without tests?

Reply:
By Spark change, you mean let's say Hudi adds another additional property? Can you give an example of what kind of Spark changes would be helpful to detect for the tests?

Reply:
> By Spark change, you mean let's say Hudi adds another additional property?

Yes. Please note that the test purpose is verifying **_compatibility_** between Trino and Spark. Just testing Trino doesn't verify that.

Reply:
@ebyhr I added an assertion for Spark but it is failing because the pattern does not match (need to assert pattern because one of the table properties is dependent on the commit time which is not fixed). Can you please check the latest commit? 
```
tests               | java.lang.AssertionError:
tests               | Expecting:
tests               |  "CREATE TABLE `default`.`test_hudi_cow_show_create_egwpssb7iuq9` (
tests               |   `_hoodie_commit_time` STRING,
tests               |   `_hoodie_commit_seqno` STRING,
tests               |   `_hoodie_record_key` STRING,
tests               |   `_hoodie_partition_path` STRING,
tests               |   `_hoodie_file_name` STRING,
tests               |   `id` BIGINT,
tests               |   `name` STRING,
tests               |   `price` INT,
tests               |   `ts` BIGINT)
tests               | USING hudi
tests               | LOCATION 's3://trino-ci-test/test_hudi_cow_show_create_egwpssb7iuq9'
tests               | TBLPROPERTIES (
tests               |   'primaryKey' = 'id',
tests               |   'last_commit_time_sync' = '20221116144805187',
tests               |   'type' = 'cow',
tests               |   'preCombineField' = 'ts')
tests               | "
tests               | to match pattern:
tests               |  "\QCREATE TABLE `default`.`test_hudi_cow_show_create_egwpssb7iuq9` (
tests               |   `_hoodie_commit_time` STRING,
tests               |   `_hoodie_commit_seqno` STRING,
tests               |   `_hoodie_record_key` STRING,
tests               |   `_hoodie_partition_path` STRING,
tests               |   `_hoodie_file_name` STRING,
tests               |   `id` BIGINT,
tests               |   `name` STRING,
tests               |   `price` INT,
tests               |   `ts` BIGINT)
tests               | USING hudi
tests               | LOCATION 's3://trino-ci-test/test_hudi_cow_show_create_egwpssb7iuq9'
tests               | TBLPROPERTIES (
tests               |   'primaryKey' = 'id',
tests               |   'last_commit_time_sync' = '\d+',
tests               |   'type' = 'cow',
tests               |   'preCombineField' = 'ts')\"
tests               | 	at io.trino.tests.product.hudi.TestHudiSparkCompatibility.testCopyOnWriteShowCreateTable(TestHudiSparkCompatibility.java:78)
```

Reply:
You can try:
```java
            String lastCommitTimeSync = (String) onHudi().executeQuery("show TBLPROPERTIES " + tableName + " ('last_commit_time_sync')").project(2).getOnlyValue();
            Assertions.assertThat((String) onHudi().executeQuery("SHOW CREATE TABLE default." + tableName).getOnlyValue())
                    .isEqualTo(format("""
                                    CREATE TABLE `default`.`%s` (
                                      `_hoodie_commit_time` STRING,
                                      `_hoodie_commit_seqno` STRING,
                                      `_hoodie_record_key` STRING,
                                      `_hoodie_partition_path` STRING,
                                      `_hoodie_file_name` STRING,
                                      `id` BIGINT,
                                      `name` STRING,
                                      `price` INT,
                                      `ts` BIGINT)
                                    USING hudi
                                    LOCATION 's3://%s/%s'
                                    TBLPROPERTIES (
                                      'primaryKey' = 'id',
                                      'last_commit_time_sync' = '%s',
                                      'type' = 'cow',
                                      'preCombineField' = 'ts')
                                    """,
                            tableName,
                            bucketName,
                            tableName,
                            lastCommitTimeSync));
```